exercises/047_methods.zig

```diff
@@ -2,41 +2,44 @@
 // Help! Evil alien creatures have hidden eggs all over the Earth
 // and they're starting to hatch!
 //
-// Before you jump into battle, you'll need to know four things:
+// Before you jump into battle, you'll need to know three things:
 //
-// 1. You can attach functions to structs:
+// 1. You can attach functions to structs (and other "type definitions"):
 //
 //     const Foo = struct{
 //         pub fn hello() void {
 //             std.debug.print("Foo says hello!\n", .{});
 //         }
 //     };
 //
-// 2. A function that is a member of a struct is a "method" and is
-//    called with the "dot syntax" like so:
+// 2. A function that is a member of a struct is "namespaced" within
+//    that struct and is called by specifying the "namespace" and then
+//    using the "dot syntax":
 //
 //     Foo.hello();
 //
-// 3. The NEAT feature of methods is the special parameter named
-//    "self" that takes an instance of that type of struct:
+// 3. The NEAT feature of these functions is that if they take either
+//    an instance of the struct or a pointer to an instance of the struct
+//    then they have some syntax sugar:
 //
 //     const Bar = struct{
-//         number: u32,
-//
-//         pub fn printMe(self: Bar) void {
-//             std.debug.print("{}\n", .{self.number});
-//         }
+//         pub fn a(self: Bar) void { _ = self; }
+//         pub fn b(this: *Bar, other: u8) void { _ = this; _ = other; }
+//         pub fn c(bar: *const Bar) void { _ = bar; }
```

Comment:
These are good examples. But I don't believe exercises `001` - `046` have used the `_` placeholder much and I'm afraid this is going to look a bit alien compared to what has come before and distract from the point being made. (I might be remembering wrong. Maybe I used them more than I think.)

Even though it will take up more room, I think I'd like to see all three of those examples expanded out (could just be a `print()` example in the body of each.

Reply:
Yea, that makes sense.

I think I could also just remove the `_`? It does mean that they would no longer compile, but I don't think that's necessary here?

I can try to make them more interesting (still only one or two line) examples that `a`, `b`, `c`.